### ISSUE TITLE
RichFilemanager upload event

### ISSUE THREAD

**Issue by supun19** [servocoder/RichFilemanager#275]
In my case: I configured RichFilemanager with S3 bucket. I have another application which can upload file to S3 bucket. So I need a mechanism when I upload file to S3 bucket through other application the RichFilemanger would be known. I found few solution(pub/sub solution).
So I need rich file manager to refresh the listing of files, when I upload a file to S3 using my other application.
1. S3 bucket can subscribe Amazon Simple Notification Service. So If we could  implement mechanism to subscribe ASN in RichFilemanger. 

2. S3 can trigger lambda function. In this scenario we have own socket server. RichFilemanager subscribed to a socket server. When S3 trigger event to run a lambda function  It notifies the socket server. Then socket server push message to RichFilemanager.

****I would like to know whether RichFilemanger has event mechanism to update 
  where  when receive a update massage to refresh browser(read whole folder again because new file are received)

**Comment by servocoder** [comment]
This feature is easy to implement if you are talking about the client-side method that will update (refresh) the currently open folder with a server request. 

RichFilemanager already has public methods mechanism, for example you can try this one at the demo page:

```javascript
$('.fm-container').data('richFilemanager').error('Test error');
```

I can add a method to reload the currently open folder.

Let me know if this is what you want.

**Comment by supun19** [comment]
Yes, that is exactly what I'm looking for

**Comment by servocoder** [comment]
Added 2 public plugin methods.

Use this one to reload the currently viewed folder:

`$('.fm-container').data('richFilemanager').refreshFolder(applyTreeNode);`

And this one is to load content of specified relative folder path:

`$('.fm-container').data('richFilemanager').loadFolder(relativePath, applyTreeNode);`

The `applyTreeNode` argument expects boolean value and defines whether appropriate filetree node should be reloaded (adjusted) as well. This will take effect only if the filetree node exists and visible.

This feature affected some core methods, so let me know if you find that something is broken, please.

**Comment by supun19** [comment]
Thank You Very much and If i get issue i will let you

**Comment by supun19** [comment]
no any issue it is work as I expect

**Comment by servocoder** [comment]
Released in v2.7.1
